app/routes/projects/components/resources/index.tsx

```diff
@@ -0,0 +1,159 @@
+import { zfd } from "zod-form-data";
+import { withZod } from "@remix-validated-form/with-zod";
+import { z } from "zod";
+import SimpleAutocompleteField from "~/core/components/SimpleAutocompleteField";
+import { useFieldArray, ValidatedForm } from "remix-validated-form";
+
+const RESOURCE_TYPES = [
+  "Cloud Account",
+  "Domain",
+  "Hardware (cellphone, console)",
+  "License",
+  "Other",
+];
+
+const RESOURCE_PROVIDERS = ["AWS", "GCP", "Azure"];
+const RESOURCE_NAMES: string[] = [];
+
+interface IResource {
+  type: string;
+  provider: string;
+  name: string;
+}
+
+interface IProps {
+  allowEdit: Boolean;
+  projectResources: IResource[];
+  resourceData: { types: string[]; providers: string[]; names: string[] };
+}
+
+export const validator = withZod(
+  zfd.formData({
+    resources: z.array(
+      z.object({
+        type: zfd.text(),
+        provider: zfd.text(),
+        name: zfd.text(),
+      })
+    ),
+  })
+);
+
+export default function Resources({
+  allowEdit = false,
+  projectResources,
+  resourceData,
+}: IProps) {
+  const transition = useTransition();
+  const [isEditActive, setIsEditActive] = useState(false);
+  const toggleChangeEditView = () => setIsEditActive((prevValue) => !prevValue);
+  const [resources, { push: addResource, remove: removeResource }] =
+    useFieldArray("resources", { formId: "projectResourcesForm" });
+
+  const resourceTypes = [...new Set(RESOURCE_TYPES.concat(resourceData.types))];
+  const resourceProviders = [
+    ...new Set(RESOURCE_PROVIDERS.concat(resourceData.providers)),
+  ];
+  const resourceNames = [...new Set(RESOURCE_NAMES.concat(resourceData.names))];
+
+  return (
+    <Card>
+      <ValidatedForm
+        method="post"
+        id="projectResourcesForm"
+        validator={validator}
+        subaction="UPDATE_RESOURCES"
+        defaultValues={{ resources: projectResources }}
+      >
+        <CardHeader
+          title="Resources:"
+          action={
+            allowEdit &&
+            (isEditActive ? (
+              <IconButton type="reset">
+                <Close>Cancel</Close>
+              </IconButton>
+            ) : (
+              <IconButton onClick={() => toggleChangeEditView()}>
+                <EditSharp></EditSharp>
+              </IconButton>
+            ))
+          }
+        />
+        <CardContent>
+          {isEditActive && (
+            <Button
+              disabled={transition.state === "submitting"}
+              variant="contained"
+              type="button"
+              sx={{
+                position: "absolute",
+                marginTop: "-67px",
+                marginLeft: "150px",
+              }}
+              onClick={() => {
+                addResource({ type: resourceTypes[0], provider: "", name: "" });
+              }}
+            >
+              Add new resource
+            </Button>
+          )}
+          {resources.map((resource, index) => (
+            <ResourceRow key={index}>
```

Comment:
This could be replaced with MUI's <Grid> or <Stack> components, to have one less file to maintain (the .styles file).

Reply:
Fixed